extensions/cache/runtime/src/main/java/io/quarkus/cache/runtime/caffeine/CaffeineCacheImpl.java

```diff
@@ -122,6 +124,26 @@ public CompletableFuture<Object> apply(Object key) {
                 }).map(fromCacheValue());
     }
 
+    @Override
+    public <K, V> V compute(K key, BiFunction<K, V, V> command) {
+        Objects.requireNonNull(key, NULL_KEYS_NOT_SUPPORTED_MSG);
+        ConcurrentMap<Object, Object> cacheMap = cache.synchronous().asMap();
+        Object previous = cacheMap.get(key);
+        V computed = command.apply(key, cast(NullValueConverter.fromCacheValue(previous)));
+
+        Object computedCacheValue = NullValueConverter.toCacheValue(computed);
+        boolean replaced;
+        if (previous == null) {
+            replaced = cacheMap.putIfAbsent(key, computedCacheValue) == null;
+        } else {
+            replaced = cacheMap.replace(key, previous, computedCacheValue);
+        }
+        if (!replaced) {
+            throw new RuntimeException("Concurrent modification, cache has not been modified");
+        }
+        return computed;
+    }
```

Comment:
@ben-manes this is the compute method I am trying to implement on top of Caffeine. it is similar to the compute method in the caffeine api without the retry